packages/authentication/docs/authentication-system.md

```diff
@@ -163,197 +163,3 @@ And the abstractions for:
-    const credentials: BasicAuthenticationStrategyCredentials = this.extractCredentials(
-      request,
-    );
-    const user = await this.userService.verifyCredentials(credentials);
-    const userProfile = this.userService.convertToUserProfile(user);
-
-    return userProfile;
-  }
-```
-
-A custom sequence must be created to insert the
-`AuthenticationBindings.AUTH_ACTION` action. The `AuthenticateFn` function
-interface is implemented by the `value()` function of
-`AuthenticateActionProvider` class in `/src/providers/auth-action.provider.ts`.
-
-```ts
-export class SequenceIncludingAuthentication implements SequenceHandler {
-  constructor(
-    @inject(SequenceActions.FIND_ROUTE) protected findRoute: FindRoute,
-    @inject(SequenceActions.PARSE_PARAMS)
-    protected parseParams: ParseParams,
-    @inject(SequenceActions.INVOKE_METHOD) protected invoke: InvokeMethod,
-    @inject(SequenceActions.SEND) protected send: Send,
-    @inject(SequenceActions.REJECT) protected reject: Reject,
-    @inject(AuthenticationBindings.AUTH_ACTION)
-    protected authenticateRequest: AuthenticateFn,
-  ) {}
-
-  async handle(context: RequestContext) {
-    try {
-      const {request, response} = context;
-      const route = this.findRoute(request);
-
-      //call authentication action
-      await this.authenticateRequest(request);
-
-      // Authentication successful, proceed to invoke controller
-      const args = await this.parseParams(request, route);
-      const result = await this.invoke(route, args);
-      this.send(response, result);
-    } catch (error) {
-      //
-      // The authentication action utilizes a strategy resolver to find
-      // an authentication strategy by name, and then it calls
-      // strategy.authenticate(request).
-      //
-      // The strategy resolver throws a non-http error if it cannot
-      // resolve the strategy. When the strategy resolver obtains
-      // a strategy, it calls strategy.authenticate(request) which
-      // is expected to return a user profile. If the user profile
-      // is undefined, then it throws a non-http error.
-      //
-      // It is necessary to catch these errors and add HTTP-specific status
-      // code property.
-      //
-      // Errors thrown by the strategy implementations already come
-      // with statusCode set.
-      //
-      // In the future, we want to improve `@loopback/rest` to provide
-      // an extension point allowing `@loopback/authentication` to contribute
-      // mappings from error codes to HTTP status codes, so that application
-      // don't have to map codes themselves.
-      if (
-        error.code === AUTHENTICATION_STRATEGY_NOT_FOUND ||
-        error.code === USER_PROFILE_NOT_FOUND
-      ) {
-        Object.assign(error, {statusCode: 401 /* Unauthorized */});
-      }
-
-      this.reject(context, error);
-      return;
-    }
-  }
-}
-```
-
-Then custom sequence must be bound to the application, and the authentication
```

Comment:
Is this following part covered in the new docs? I couldn't find `addExtension` in https://loopback.io/doc/en/lb4/Loopback-component-authentication.html

Reply:
@nabdelgadir 

This is old and was replaced with `registerAuthenticationStrategy`

Reply:
https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/types.ts#L65

```ts
export function registerAuthenticationStrategy(
  context: Context,
  strategyClass: Constructor<AuthenticationStrategy>,
) {
  addExtension(
    context,
    AuthenticationBindings.AUTHENTICATION_STRATEGY_EXTENSION_POINT_NAME,
    strategyClass,
    {
      namespace:
        AuthenticationBindings.AUTHENTICATION_STRATEGY_EXTENSION_POINT_NAME,
    },
  );
}

```

Reply:
https://loopback.io/doc/en/lb4/Loopback-component-authentication.html#registering-a-custom-authentication-strategy

Reply:
Thanks!